app/src/main/java/com/globant/practice/presentation/presenter/WebClientPresenter.java

```diff
@@ -0,0 +1,77 @@
+package com.globant.practice.presentation.presenter;
+
+import com.globant.practice.presentation.model.WebClientState;
+import com.globant.practice.presentation.view.fragment.WebClientView;
+import javax.inject.Inject;
+
+/**
+ * Manages the possible states of the WebClientFragment using a view model called WebClientState,
+ * load the web page on the WebView and shows o dismiss the progress dialog on the WebClientFragment
+ * Created by jonathan.vargas on 4/05/2017.
+ */
+
+public class WebClientPresenter extends BasePresenter<WebClientView> {
+    private WebClientState webClientState;
+
+    /**
+     * Construct method of the WebClientPresenter and initializes the WebClientState model
+     */
+    @Inject
+    public WebClientPresenter() {
+        webClientState = new WebClientState();
+    }
+
+    /**
+     * Initializes the web page load progress
+     *
+     * @param htmlUrl    address of the web page that will be load
+     * @param detailType indicates if the web page are a profile or a repository
+     */
+    public void loadWebPage(String htmlUrl, String detailType) {
+        if (!webClientState.isLoading()) {
+            webClientState.setHtmlUrl(htmlUrl);
+            webClientState.setDetailType(detailType);
+            webClientState.setLoading(true);
+            webClientState.setErrorShowing(false);
+            if (isViewAttached()) {
```

Comment:
Before to changed the call to the presenter.detachView to the onPause state when the user changed the orientation the app makes 4 calls to the webView.loadUrl, with this condition I got that only makes two calls, but now with the call of the presenter.detachView on the onPause state this condition is useless, I'm will erase this condition and see how I can control that the app only make one call to the webView.loadUrl